pylint/checkers/imports.py

```diff
@@ -555,13 +555,18 @@ def visit_importfrom(self, node: nodes.ImportFrom) -> None:
             self._check_position(node)
         if isinstance(node.scope(), nodes.Module):
             self._record_import(node, imported_module)
-        if imported_module is None:
+
+        # If we can't find the import we still try and build the import graph with our
+        # best guess: the relative name of the import.
+        imported_module_name = imported_module.name if imported_module else node.modname
+        if imported_module_name is None:
             return
+
         for name, _ in node.names:
             if name != "*":
-                self._add_imported_module(node, f"{imported_module.name}.{name}")
+                self._add_imported_module(node, f"{imported_module_name}.{name}")
             else:
-                self._add_imported_module(node, imported_module.name)
+                self._add_imported_module(node, imported_module_name)
```

Comment:
Seems like something that could be it's own PR.

Reply:
Yeah and it is breaking tests... Without it this test is very flaky so I'm not sure what to do